core/src/client/mod.rs

```diff
@@ -675,4 +703,540 @@ impl StarknetClient for StarknetClientImpl {
+
+                        ether_tx.nonce = felt_to_u256(v0.nonce);
+                        ether_tx.from = starknet_address_to_ethereum_address(v0.contract_address);
+                        // Define gas_price data
+                        ether_tx.gas_price = None;
+                        // Extracting the signature
+                        ether_tx.r = felt_option_to_u256(v0.signature.get(0))?;
+                        ether_tx.s = felt_option_to_u256(v0.signature.get(1))?;
+                        ether_tx.v = felt_option_to_u256(v0.signature.get(2))?;
+                        // Extracting the data (transform from calldata)
+                        ether_tx.input = vec_felt_to_bytes(v0.calldata);
+                        //TODO:  Fetch transaction To
+                        ether_tx.to = None;
+                        //TODO:  Fetch value
+                        ether_tx.value = U256::from(100);
+                        //TODO: Fetch Gas
+                        ether_tx.gas = U256::from(100);
+                        // Extracting the chain_id
+                        ether_tx.chain_id = Some(CHAIN_ID.into());
+                        // Extracting the standard_v
+                        ether_tx.standard_v = U256::from(0);
+                        // Extracting the creates
+                        ether_tx.creates = None;
+                        // How to fetch the public_key?
+                        ether_tx.public_key = None;
+                        // ...
+                        ether_tx.block_hash = block_hash;
+                        ether_tx.block_number = block_number;
+                    }
+
+                    InvokeTransaction::V1(v1) => {
+                        // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+
+                        ether_tx.hash =
+                            PrimitiveH256::from_slice(&v1.transaction_hash.to_bytes_be());
+                        class_hash = self
+                            .client
+                            .get_class_hash_at(
+                                &StarknetBlockId::Tag(BlockTag::Latest),
+                                v1.sender_address,
+                            )
+                            .await?;
+
+                        ether_tx.nonce = felt_to_u256(v1.nonce);
+                        ether_tx.from = starknet_address_to_ethereum_address(v1.sender_address);
+                        // Define gas_price data
+                        ether_tx.gas_price = None;
+                        // Extracting the signature
+                        ether_tx.r = felt_option_to_u256(v1.signature.get(0))?;
+                        ether_tx.s = felt_option_to_u256(v1.signature.get(1))?;
+                        ether_tx.v = felt_option_to_u256(v1.signature.get(2))?;
+                        // Extracting the data
+                        ether_tx.input = vec_felt_to_bytes(v1.calldata);
```

Comment:
We can add a TODO to properly parse the data

Reply:
Ok!

Reply:
Could you create an issue for it?

currently v1.calldata is a callarray as per https://twitter.com/ETazou/status/1624078157337751553